integration/ledger/snapshot_test.go

```diff
@@ -1005,3 +1028,100 @@ func invokeWithoutPassingOrdererEndPoint(n *nwo.Network, peer *nwo.Peer, channel
 	invokeChaincode(n, peer, command)
 	nwo.WaitUntilEqualLedgerHeight(n, channelID, nwo.GetLedgerHeight(n, peer, channelID), n.PeersWithChannel(channelID)...)
 }
+
+// commitTx commits a transaction for a given transaction envelope
+func commitTx(n *nwo.Network, orderer *nwo.Orderer, peer *nwo.Peer, channelID string, env *cb.Envelope, txid string) error {
+	var cancelFunc context.CancelFunc
+	ctx, cancelFunc := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancelFunc()
+
+	// get signing identity
+	By("getting default SigningIdentity for user1 on peer " + peer.ID())
+	var mspID string
+	for _, org := range n.Organizations {
+		if org.Name == peer.Organization {
+			mspID = org.MSPID
+			break
+		}
+	}
+	localMSP, err := loadLocalMSPAt(n.PeerUserMSPDir(peer, "User1"), mspID, "bccsp")
+	Expect(err).To(BeNil())
+	signer, err := localMSP.GetDefaultSigningIdentity()
+	Expect(err).To(BeNil())
+
+	// create deliver client and delivergroup
+	peerClient := &common.PeerClient{
+		CommonClient: common.CommonClient{
+			GRPCClient: grpcClient(filepath.Join(n.PeerLocalTLSDir(peer), "ca.crt")),
+			Address:    n.PeerAddress(peer, nwo.ListenPort),
+		},
+	}
+	deliverClient, err := peerClient.PeerDeliver()
+	Expect(err).NotTo(HaveOccurred())
+	dg := ic.NewDeliverGroup([]pb.DeliverClient{deliverClient}, []string{n.PeerAddress(peer, nwo.ListenPort)}, signer, tls.Certificate{}, channelID, txid)
+	err = dg.Connect(ctx)
+	Expect(err).NotTo(HaveOccurred())
+
+	// create orderer client and broadcast client
+	By("creating orderer client to send transaction to the orderer" + peer.ID())
+	ordererClient := &common.OrdererClient{
+		CommonClient: common.CommonClient{
+			GRPCClient: grpcClient(filepath.Join(n.OrdererLocalTLSDir(orderer), "ca.crt")),
+			Address:    n.OrdererAddress(orderer, nwo.ListenPort),
+		},
+	}
+	bc, err := ordererClient.Broadcast()
+	Expect(err).NotTo(HaveOccurred())
+	broadcastClient := &common.BroadcastGRPCClient{Client: bc}
+	err = broadcastClient.Send(env)
+	Expect(err).NotTo(HaveOccurred())
+
+	// wait for deliver event
+	By("waiting for deliver event on peer " + peer.ID())
+	return dg.Wait(ctx)
+}
+
+func loadLocalMSPAt(dir, id, mspType string) (msp.MSP, error) {
```

Comment:
Here is an [easier way](https://github.com/hyperledger/fabric/blob/master/integration/raft/cft_test.go#L895-L901) to create a signer than to make a local MSP.

Reply:
Thank you Yacov. Done.